class/DatabaseConnection.php

```diff
@@ -0,0 +1,340 @@
+<?php
+
+require_once('config/error_messages.php');
+require_once('config/constants.php');
+require_once('error_helper.php');
+
+/**
+* Super Class
+*
+* @author Smruti Ranjan
+*/
+class DatabaseConnection
+{
+	public static $conn = NULL;
+	public static $connection_obj = NULL;
+
+	/**
+	* Constructor to create a database connection
+	*
+	* @access public
+	* @param  void
+	* @return void[]
+	*/
+	private function __construct($db_param)
+	{
+		self::$conn = mysqli_connect($db_param['hostname'], $db_param['username'], 
+			$db_param['password'], $db_param['database']);
+
+		if(mysqli_connect_errno(self::$conn))
+		{
+			log_error('Creation of databse connection');
+			die ('Failed to connect to Database, Please try later :' . mysqli_connect_error());
+		}
+	}
+
+	/**
+	* To check of exixting connection object
+	*
+	* @access public
+	* @param  void
+	* @return object
+	*/
+	public static function create_connection($db_param)
+	{
+		if(self::$connection_obj == NULL)
+		{
+			self::$connection_obj = new DatabaseConnection($db_param);
+		}
+
+		return self::$connection_obj;
+	}
+
+	/**
+	* To execute SQL queries
+	*
+	* @access public
+	* @param  string $query
+	* @return object
+	*/
+	public static function db_query($query)
+	{
+		$result = mysqli_query(self::$conn, $query);
+		return $result;
+	}
+
+	/**
+	* To fetch executed select query
+	*
+	* @access public
+	* @param  object $passed_object
+	* @return object
+	*/
+	public static function db_fetch_array($passed_object)
```

Comment:
Can use better variable name like '$result_obj' or '$result_object' in place of $passed_object